src/Collateral.sol

```diff
@@ -27,14 +27,19 @@ contract Collateral {
     struct CollateralAccount {
         uint256 balance; // Total collateral balance for a sender-receiver pair
         uint256 amountThawing; // Amount of collateral currently being thawed
-        uint256 thawEndTimestamp; // Block number at which thawing period ends
+        uint256 thawEndTimestamp; // Block number at which thawing period ends (zero if not thawing)
+    }
+
+    struct AuthorizedSigner {
```

Comment:
I think the struct name is slightly confusing here. Shouldn't it reference the sender instead of the signer? Perhaps `SenderAuthorization` or something along that vein?